lib/internal/http2/core.js

```diff
@@ -1399,20 +1399,19 @@ function trackWriteState(stream, bytes) {
   session[kHandle].chunksSentSinceLastWrite = 0;
 }
 
-function afterDoStreamWrite(status, handle, req) {
+function afterDoStreamWrite(status, handle) {
   const stream = handle[kOwner];
   const session = stream[kSession];
 
   stream[kUpdateTimer]();
 
-  const { bytes } = req;
+  const { bytes } = this;
   stream[kState].writeQueueSize -= bytes;
 
   if (session !== undefined)
     session[kState].writeQueueSize -= bytes;
-  if (typeof req.callback === 'function')
-    req.callback(null);
-  req.handle = undefined;
```

Comment:
This makes me wonder, is `handle` on `WriteWrap` used at all? I had a brief look at the C++ side and saw nothing. All I'm seeing is it being set/deleted all around. The only place I see it being used at all is `process_wrap`.

Admittedly I could be missing something more intricate here...

Reply:
@apapirovski I think it’s only for diagnostic purposes and preventing the handle from being garbage collected; I think at least in the case of `JSStream`s that could happen because that’s a weak handle and otherwise there might not be any backreference to it?

Reply:
Trying to think about this... Since the socket & http2 implementations reference the handle on `_handle` and `kHandle`, this seems like it would mainly come into play if the stream is socket/session is destroyed? Would the handle still be needed in that case? I don't recall if this would still trigger `oncomplete` or not.

Reply:
I figured no harm in putting this to practice... So far, at least as far as http2 is concerned, it seems like removing `.handle` on WriteWrap & ShutdownWrap is fine, even when running stress tests. I might play around with explicit `global.gc()` calls and study the code in more detail to understand how useful these references are.

Reply:
@apapirovski I can’t think of any way in which this would break HTTP/2, yes…

I’m a bit worried removing it might break async_hooks users… but then again, this really isn’t supposed to be public API. :/